nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/test/java/org/apache/nifi/snmp/helper/testrunners/SNMPTestRunners.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.snmp.helper.testrunners;
```

Comment:
I think the same applies here as above: the downside of adding a lot of util classes into the tests (like a lot of boilerplate, extra abstractions and fragmented tests) overweights the benefits.

Reply:
I completely disagree. The extra utility is worth it, spares lots of lines of boilerplate code in test classes and lets us to write tests way easier then creating configurations and test runners directly by hand.

Reply:
I reduced the complexity by removing the unnecessary factory providers but I'll leave the utility classes like this. The new parameterized tests also make use of it.